### PR TITLE
Double sketches

### PR DESCRIPTION
**Pull request by cosmic-chichu**

### CODE REVIEW

File: builder/postaggregation/post_aggregator.go [hunk 1]
```diff
@@ -62,6 +62,8 @@ func Load(data []byte) (builder.PostAggregator, error) {
 		p = NewLongLeast()
 	case "quantilesFromTDigestSketch":
 		p = NewQuantilesFromTDigestSketch()
+	case "quantilesDoublesSketchToQuantile":
```
Comment (vigith) [inline builder/postaggregation/post_aggregator.go:65]: TDigest uses `from` v/s DoublesSketch is using `To`? can we use the same naming convention?

**Review by vigith — CHANGES_REQUESTED** [review]

File: builder/postaggregation/post_aggregator.go (hunk 1 above)
Comment (vigith) [reply builder/postaggregation/post_aggregator.go:65]: it is based on the type specified here: https://druid.apache.org/docs/latest/development/extensions-core/datasketches-quantiles.html#quantile

**Review by vigith — DISMISSED** [review]

File: builder/postaggregation/quantiles_doubles_sketch_to_quantile.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// SetField set QuantilesDoublesSketchToQuantile
+func (q *QuantilesDoublesSketchToQuantile) SetField(field *QuantilesDoublesSketchToQuantileField) *QuantilesDoublesSketchToQuantile {
+	q.Field = field
+	return q
+}
+
+func NewQuantilesDoublesSketchToQuantileField() *QuantilesDoublesSketchToQuantileField {
+	qf := &QuantilesDoublesSketchToQuantileField{}
+	return qf
```
Comment (jbguerraz) [inline builder/postaggregation/quantiles_doubles_sketch_to_quantile.go:45]: maybe `return &QuantilesDoublesSketchToQuantileField{}` to save one useless assignation

File: examples/doubkesSketchExample.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	"github.com/grafadruid/go-druid/builder/granularity"
+	"github.com/grafadruid/go-druid/builder/intervals"
+	"github.com/grafadruid/go-druid/builder/postaggregation"
+	"github.com/grafadruid/go-druid/builder/query"
+	"log"
+	"time"
+)
+
+func getConnection() *druid.Client {
+	d, err := druid.NewClient("https://getafix-qal.odldruid-ppd.a.intuit.com")
```
Comment (jbguerraz) [inline examples/doubkesSketchExample.go:23]: probably better to replace it with `http://localhost:8888`

File: examples/doubkesSketchExample.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+/*
+	Prerequisite: These examples will work only if you have tdigest sketch data in your datastore.
+	To experiment, you can use the doubles_sketch_data.tsv file attached in this repo. It is a copy of  https://github.com/apache/druid/blob/master/extensions-contrib/tdigestsketch/src/test/resources/doubles_sketch_data.tsv
+*/
+
+// tdigestSketchUsingBuilder example using Builder Pattern
+func doublesSketchUsingBuilder() {
+	d := getConnection()
+	table := datasource.NewTable().SetName("o11y-gsapi-apigw")
```
Comment (jbguerraz) [inline examples/doubkesSketchExample.go:43]: maybe change the table name for something more generic ? like `double-sketch` or so ?

File: builder/aggregation/quantiles_doubles_sketch.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+// https://druid.apache.org/docs/latest/development/extensions-core/datasketches-quantiles.html
+type QuantilesDoublesSketch struct {
+	Base
+	FieldName string `json:"fieldName, omitempty"`
+	K         int64  `json:"k, omitempty"`
+}
+
+// NewQuantilesDoublesSketch create a new instance of QuantilesDoublesSketch with type QuantilesDoublesSketch
+func NewQuantilesDoublesSketch() *QuantilesDoublesSketch {
+	t := &QuantilesDoublesSketch{}
```
Comment (jbguerraz) [inline builder/aggregation/quantiles_doubles_sketch.go:14]: why `t` ? why not `q` instead ?

**Review by jbguerraz — CHANGES_REQUESTED** [review]
Nice work @cosmic-chichu  :) thank you!

File: examples/doublesSketchExample.go [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+			"queryType": "groupBy",
+			"dataSource": {
+				"type": "table",
+				"name": "double-sketch"
+			},
+			"granularity": "ALL",
+			"dimensions": [{
+			  "type": "default",
+			  "dimension": "uniqueId"
+    		}],
```
Comment (jbguerraz) [inline examples/doublesSketchExample.go:98]: nitpick but wrong indent

**Review by jbguerraz — CHANGES_REQUESTED** [review]
small nitpick

**Review by jbguerraz — APPROVED** [review]
LGTM! Thank you again :)

**Pull request merged**